### PR TITLE
feat: create Custom Carousel component for CardsDs

### PR DESCRIPTION
**Pull request by michael-odonovan**
#### What is it doing?
Tell us what it is doing from a technical perspective.
Creates custom Carousel component for CardsDs in CL.

#### Why is this required?
Tell us why this is required from a business/product perspective.
UX improvements designed by Curtis.

#### link to Jira ticket:
Add a link to the Jira ticket.
https://comicrelief.atlassian.net/browse/ENG-3384

### Quick Checklist:
- [x] My PR title follows the Conventional Commit spec.

- [x] I have filled out the PR description as per the template above.

- [x] I have added tests to cover new or changed behaviour.

- [x] I have updated any relevant documentation.

### CODE REVIEW

**Comment by AndyEPhipps** [comment]
> I tried the usual importing of CL commit but spent literally all day on Friday trying to get it to work. I can't see what I am doing wrong but the import commit is throwing errors around components not being able to find the CL... Bit tired of debugging so if we could just merge this cards carousel work at this basic functional stage and then I will come back and polish again after I see it in situ in a CR.com PR. There is no CardsCarousel effect on master at moment.

Hey man, @zhibek  made an update to CL during the initial Donate sprint to make it more lightweight, and the knock-on effect is that it's no longer building the `/dist` folder during the PR build process; something that we'd previously been able to use in these contexts. 

I got pulled into other things, so ended-up sidestepping it for CRcom work I was having to rush through by un-gitignoring the dist folder in my CL PR, locally building and pushing the changes to it (see [here](https://github.com/comicrelief/component-library/blob/master/.gitignore#L13)).  **However**, this is a royal pain in the arse, so wouldn't recommend doing it, as it requires having to revert those gitignore changes before that CL PR can be merged... plus it's just yet more faffing.

I think it'd be beneficial to pick over John's update and see if we're able to walk _some_ of it back to allow the usual 'test CL PR branch in CRcom PR locally' functionality, but for the time being, we might just need to release stuff to the CL a bit more often.

This only _really_ becomes dangerous if multiple people are working in the CL and CRcom at once (since I could potentially push live an update to your work that ends up needing more development). I think I'm done working within CR and CL for the short term, so happy to have this merged in, to let you test within your local CR. If it ends up needing another CL release, so be it.

File: src/components/Molecules/CardDs/CardDs.style.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardDs/CardDs.style.js:22]: Could we use our zIndex function here instead to keep it within that context, presuming its max index of 4 is enough?

File: src/components/Molecules/CardDs/CardDs.style.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardDs/CardDs.style.js:89]: Ditto here

**Comment by AndyEPhipps** [comment]
Seems like this has introduced a layout change?

Currently in `master`:
Screenshot 2024-09-10 at 10 44 21

Your updates:
Screenshot 2024-09-10 at 10 44 13

File: src/components/Molecules/CardsCarousel/CardsCarousel.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardsCarousel/CardsCarousel.js:1]: Does this still need addressing?

File: src/components/Molecules/CardsCarousel/_CarouselButtons.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardsCarousel/_CarouselButtons.js:1]: Can we fix the underlying problem to avoid having to do this?

**Review by AndyEPhipps — CHANGES_REQUESTED** [review]

**Comment by zhibek** [comment]
> > I tried the usual importing of CL commit but spent literally all day on Friday trying to get it to work. I can't see what I am doing wrong but the import commit is throwing errors around components not being able to find the CL... Bit tired of debugging so if we could just merge this cards carousel work at this basic functional stage and then I will come back and polish again after I see it in situ in a CR.com PR. There is no CardsCarousel effect on master at moment.
> 
> Hey man, @zhibek made an update to CL during the initial Donate sprint to make it more lightweight, and the knock-on effect is that it's no longer building the `/dist` folder during the PR build process; something that we'd previously been able to use in these contexts.
> 
> I got pulled into other things, so ended-up sidestepping it for CRcom work I was having to rush through by un-gitignoring the dist folder in my CL PR, locally building and pushing the changes to it (see [here](https://github.com/comicrelief/component-library/blob/master/.gitignore#L13)). **However**, this is a royal pain in the arse, so wouldn't recommend doing it, as it requires having to revert those gitignore changes before that CL PR can be merged... plus it's just yet more faffing.
> 
> I think it'd be beneficial to pick over John's update and see if we're able to walk _some_ of it back to allow the usual 'test CL PR branch in CRcom PR locally' functionality, but for the time being, we might just need to release stuff to the CL a bit more often.
> 
> This only _really_ becomes dangerous if multiple people are working in the CL and CRcom at once (since I could potentially push live an update to your work that ends up needing more development). I think I'm done working within CR and CL for the short term, so happy to have this merged in, to let you test within your local CR. If it ends up needing another CL release, so be it.

Hey @michael-odonovan @AndyEPhipps. Sorry to cause you hassle with the changes from #655/#657.

If you want to replicate the old `postinstall` step locally (that I had to remove when dependencies we didn't want to package in the library were moved to `devDependencies`) then you can follow the steps below:
```
# Starting from the root of whichever project you're working on, change to the component-library directory in node_modules
cd node_modules/@comicrelief/component-library/

# Run yarn install inside here (giving you both dependencies *and devDependencies*, which you need to build)
yarn install

# Run yarn build to regenerate the dist directory (like postinstall used to do for us)
yarn build
```

Note: When I tried to run the steps above just now, I found the `glob` library didn't like that I was running Node v18 (`error glob@11.0.0: The engine "node" is incompatible with this module. Expected version "20 || >=22". Got "18.20.4"`). So I had to switch to Node v20 to run the steps above successfully (`nvm use 20`).

I hope the above steps help! (if I've understood the problem you faced, correctly?)

**Comment by AndyEPhipps** [comment]
Thanks so much for that context, @zhibek; I hope that didn't come off as me _blaming_ you! Just wanted to track the work and who was involved; it was obviously done for a good reason, and this was just a side effect :)

**Comment by zhibek** [comment]
Don't worry @AndyEPhipps. I don't feel blamed. ;) Just want to help mitigate any pain caused by these changes!

File: src/components/Molecules/CardsCarousel/CardsCarousel.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardsCarousel/CardsCarousel.js:1]: Could all of the CardsCarousel stuff go into the `Organisms` folder plz?

Also, I'd guess we need some animations (auto?), and an icon or something to let the user know how to interact with it, esp if there's no automatic play scroll thing.

Also, the images don't change with the cards?

I think it's important to catchup with Curtis about the UX and interactions here asap (as the tweaks would need another review), but it looking great so far, mate 👍

File: src/components/Molecules/CardDs/CardDs.style.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardDs/CardDs.style.js:1]: Can we tweak the bottom padding in this context so it matches the top? It looks a bit squiffy, looks like it's something to do with the negative margins

Screenshot 2024-09-19 at 10 19 52

File: src/components/Molecules/CardsCarousel/CardsCarousel.js
Comment (AndyEPhipps) [inline src/components/Molecules/CardsCarousel/CardsCarousel.js:1]: Looks like we need to handle what's happening on the 1440+ breakpoint:

Screenshot 2024-09-19 at 10 25 25

**Review by AndyEPhipps — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
